central/detection/service/service_impl.go

```diff
@@ -352,6 +376,11 @@ func (s *serviceImpl) DetectDeployTimeFromYAML(ctx context.Context, req *apiV1.D
 		eCtx.ClusterID = clusterID
 	}
 
+	ns := req.GetNamespace()
+	if ns != "" {
+		eCtx.Namespace = ns
+	}
```

Comment:
Maybe a bit more compact, like that?

```suggestion
	if ns := req.GetNamespace(); ns != "" {
		eCtx.Namespace = ns
	}
```